notebooks/kkl15.jl

```diff
@@ -0,0 +1,2102 @@
+"""
+
+# ╔═╡ bf96e629-5bf3-4292-9572-0f47dff48608
+dat_subj2 = @subset(dat_subj, :rt_m < 510);
+
+# ╔═╡ 3fb2caa9-6873-42c9-8419-421d0cec2d16
+boxplot(
+	dat_subj2.CTR.refs,
+	dat_subj2.lrt_m;
+	orientation=:horizontal,
+	show_notch=true,
+	axis=(; yticks=(
+			1:4,
+			[
+				"valid cue",
+				"same obj/diff pos",
+				"diff obj/same pos",
+				"diff obj/diff pos"
+			]
+		)
+	),
+	figure=(;resolution=(800, 300)),
+)
+
+# ╔═╡ 4001cfa9-0fbe-40e3-b324-e0c1c26c6b8c
+md" A better alternative to the boxplot is often a dotplot, because it also displays subjects' condition means."
+
+# ╔═╡ 9f41d2bf-4c14-46db-9b65-f685afb19d2c
+md" **To be done** "
+
+# ╔═╡ 6fa62496-2432-4423-9e55-fd37b3f4f849
+md"""
+We can also look at correlations plots based on the four condition means. There are actually two correlation matrices which have correspondences in alternative parameterizatios of the LMM random-effec structure. One matrix is based on the four measures. If you think of the four measures as test scores, this matrix is the usual correlation matrix. The second matrix contains correlations between the Grand Mean (GM) and the three effects defined with the contrasts for the four levels of the condition factor in the next chunk. 
+
+To this end, we
+
++ use the `unstack()` command to convert data from long to wide format,
++ compute the GM and the three experimental effects.
++ plot the correlation matrix for four measures/scores, and
++ plot the correlation matrix for GM and three effects
+
+"""
+
+# ╔═╡ f50e0fff-665b-4056-993c-b6dac881785b
+begin
+	dat_subj_w = unstack(dat_subj, :Subj, :CTR, :rt_m)
+	@transform!(dat_subj_w,  :GM = (:val + :sod + :dos + :dod) ./4)
+    @transform!(dat_subj_w,  :spatial =    :sod - :val)
+	@transform!(dat_subj_w,  :object =     :dos - :sod)
+	@transform!(dat_subj_w,  :attraction = :dod - :dos)
+	describe(dat_subj_w)
+end
+
+# ╔═╡ a5226169-dd2e-4105-9569-98fd781332a4
+@df dat_subj_w StatsPlots.corrplot(cols(2:5), grid = false, compact=false)
+
+# ╔═╡ 1e740cc3-3037-4fa3-82d2-75b13b86a593
+@df dat_subj_w StatsPlots.corrplot(cols(6:9), grid = false, compact=false)
+
+# ╔═╡ 7343b5fc-3664-4f13-8a95-8795a1b0c959
+
+
+# ╔═╡ 019a4876-e32d-41f4-9636-7efaea5e4929
+md"**Note:** Two of the theoreticsally irrelevant within-subject effect correlations have a different sign than the corresponding, non-significant CPs in the LMM; they are negative here, numerically positive in the LMM. This occurs only very rarely in the case of ecological correlations. However, as they are not significant according to shortest coverage interval, it may not be that relevant either. It is the case both for effects based on log-transformed and raw reaction times."
+
+# ╔═╡ 317f7e3d-94ef-4a36-a45d-e08b5fe0cd20
+md"""## Linear mixed model
+"""
+
+# ╔═╡ e827e62a-b458-46a7-a86c-b3c3392c3e6a
+begin
+	cntr1 = Dict(
+    :CTR  => SeqDiffCoding(levels=["val", "sod", "dos", "dod"]),
```

Comment:
```suggestion
    :CTR  => SeqDiffCoding(levels=["val", "sod", "dos", "dod"]),
    :Subj => Grouping(),
```

Also, the `begin/end` is superfluous here as there is only one expression to evaluate.

Reply:
In this script, in no other script  `:Subj => Grouping()` gave me an error "Grouping" not defined. I even copied a block, restarted Julia and Pluto, etc. After a while I gave up, because I did not need it for the 86 subjects. I thought it might have been related to cells getting out of order,  but I am not sure.  It will be interesting to see whether it works after I  check out your modification.

Reply:
At a time like that you can open a temporary code window in Pluto and start experimenting to see if the Grouping function is defined in the expected place.  That is, does
```jl
MixedModels.Grouping
```
exist, and, if so, what is
```jl
julia> typeof(MixedModels.Grouping)
DataType
```

Reply:
Argh,  I was looking for `StatsModels.Grouping()`